Ogra/Encodable.swift

```diff
@@ -62,22 +62,32 @@ extension Optional where T: Encodable {
 		case .Some(let v): return v.encode()
 		}
 	}
+extension CollectionType where Generator.Element == (Key: StringLiteralConvertible, Value: Encodable) {
+    public func encode() -> JSON {
+        var values = [String : JSON]()
+        for (key, value) in self {
+            values[String(key)] = value.encode()
+        }
+        return .Object(values)
+    }
 }
 
-extension Dictionary where Key: StringLiteralConvertible, Value: Encodable {
-	public func encode() -> JSON {
-		var values = [String : JSON]()
-		for (key, value) in self {
-			values[String(key)] = value.encode()
-		}
-		return .Object(values)
-	}
+extension Optional where T: CollectionType, T.Generator.Element == (Key: StringLiteralConvertible, Value: Encodable) {
+    public func encode() -> JSON {
+        return self.map { $0.encode() } ?? .Null
+    }
```

Comment:
This optional version may need to be updated to mirror the type signature of it's counterpart.